src/main/java/edu/ie3/util/geo/GeoUtils.java

```diff
@@ -133,11 +133,12 @@ public static SortedSet<CoordinateDistance> getCoordinateDistances(
    * @param coordinates the points to calculate the distance from the base point for
    * @return a sorted set of distances between the base and other coordinates
    */
-  public static SortedSet<CoordinateDistance> calcOrderedCoordinateDistances(
+  public static Set<CoordinateDistance> calcOrderedCoordinateDistances(
       Point baseCoordinate, Collection<Point> coordinates) {
     return coordinates.stream()
         .map(coordinate -> new CoordinateDistance(baseCoordinate, coordinate))
-        .collect(Collectors.toCollection(TreeSet::new));
+        .sorted(Comparator.comparing(CoordinateDistance::getDistance))
+        .collect(Collectors.toCollection(LinkedHashSet::new));
```

Comment:
I think it would also be fine to just return a `List` here (not any `Collection` though since order needs to be preserved). We would only need a (hash)set here if we needed fast checks of containment, which I think we don't.